### PR TITLE
Initial code for better rewriting of documents

### PR DESCRIPTION
**Pull request by bbakerman**
Now working :-)

### CODE REVIEW

File: src/main/java/graphql/nadel/engine/FieldMetadataUtil.java [hunk 1]
```diff
@@ -55,14 +52,14 @@ public boolean isSkipTraversing() {
...
-        return fieldMetadata.stream().anyMatch(FieldMetadata::isSkipTraversing);
-    }
+//    public static boolean skipTraversing(Field field) {
+//        String serialized = field.getAdditionalData().get(NADEL_FIELD_METADATA);
+//        if (serialized == null) {
+//            return false;
+//        }
+//        List<FieldMetadata> fieldMetadata = readMetadata(serialized);
+//        return fieldMetadata.stream().anyMatch(FieldMetadata::isSkipTraversing);
+//    }
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/engine/FieldMetadataUtil.java]: Cleanup to be done?

Comment (andimarek) [reply src/main/java/graphql/nadel/engine/FieldMetadataUtil.java]: done

File: src/main/java/graphql/nadel/engine/transformation/RecordTypeInformation.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static graphql.Assert.assertNotNull;
+import static graphql.introspection.Introspection.TypeNameMetaFieldDef;
+import static graphql.schema.GraphQLTypeUtil.unwrapAll;
+
+public class RecordTypeInformation {
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/engine/transformation/RecordTypeInformation.java]: Should these classes go into their own top level classes.

There a fair amount of code here

Comment (andimarek) [reply src/main/java/graphql/nadel/engine/transformation/RecordTypeInformation.java]: done

Comment (bbakerman) [reply src/main/java/graphql/nadel/engine/transformation/RecordTypeInformation.java]: Also you know me - I love at least a class JavaDoc comment on what it does

**Comment by bbakerman** [comment]
TICK!

**Pull request merged**